### PR TITLE
Remove deprecated arguments

### LINKED ISSUE #1038: Remove deprecated args for v0.4.0
## 🚀 Feature

Remove some arguments that should be removed since v0.4.0.

### PR DESCRIPTION
**Pull request by erip**
Fixes #1038

Description: This aims to clean up some of the deprecated APIs that are set to be removed in 0.4.0.

Check list:
* [ ] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/engine/engine.py [hunk 1]
```diff
@@ -595,8 +589,6 @@ def run(
                 `len(data)`. If `data` is an iterator and `epoch_length` is not set, then it will be automatically
                 determined as the iteration on which data iterator raises `StopIteration`.
                 This argument should not change if run is resuming from a state.
-            seed (int, optional): Deprecated argument. Please, use `torch.manual_seed` or
```
Comment (vfdev-5) [inline ignite/engine/engine.py:598]: Seed deprecation is recent one. We should keep it until v0.5.0

File: ignite/handlers/checkpoint.py [hunk 2]
```diff
@@ -52,7 +52,6 @@ class Checkpoint:
             Input of the function is `(engine, event_name)`. Output of function should be an integer.
             Default is None, global_step based on attached engine. If provided, uses function output as global_step.
             To setup global step from another engine, please use :meth:`~ignite.handlers.global_step_from_engine`.
-        archived (bool, optional): Deprecated argument as models saved by `torch.save` are already compressed.
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:55]: Same here. Keep `archived` and please add a note that it will be removed since v0.5.0

File: ignite/handlers/checkpoint.py [hunk 3]
```diff
@@ -442,32 +430,15 @@ def __init__(
         self,
         dirname: str,
         filename_prefix: str,
-        save_interval: Optional[Callable] = None,
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:445]: keep all args, please

File: ignite/handlers/checkpoint.py [hunk 4]
```diff
@@ -442,32 +430,15 @@ def __init__(
...
         atomic: bool = True,
         require_empty: bool = True,
         create_dir: bool = True,
-        save_as_state_dict: bool = True,
         global_step_transform: Optional[Callable] = None,
-        archived: bool = False,
     ):
 
-        if not save_as_state_dict:
-            raise ValueError("Argument save_as_state_dict is deprecated and should be True")
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:458]: Add a note about removing since v0.5.0

File: ignite/handlers/checkpoint.py [hunk 5]
```diff
@@ -442,32 +430,15 @@ def __init__(
...
-        if save_interval is not None:
-            msg = (
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:460]: Same here
